dtfs-central-api/src/v1/controllers/utilisation-report-service/helpers/get-record-correction-fields.test.ts

```diff
@@ -0,0 +1,75 @@
+      .build();
+
+    it('should return the fields with formattedDateReceived and formattedBankCommentary as "-"', () => {
+      const result = getRecordCorrectionFields(feeRecordCorrectionEntity.feeRecord, feeRecordCorrectionEntity);
+
+      const { formattedOldRecords } = getFormattedOldAndCorrectValues(feeRecordCorrectionEntity, feeRecordCorrectionEntity.feeRecord);
+      const reasonsArray = mapReasonsToDisplayValues(feeRecordCorrectionEntity.reasons);
+
+      const expected = {
+        facilityId: feeRecordCorrectionEntity.feeRecord.facilityId,
+        correctionId: feeRecordCorrectionEntity.id,
+        feeRecordId: feeRecordCorrectionEntity.feeRecord.id,
+        exporter: feeRecordCorrectionEntity.feeRecord.exporter,
+        formattedReasons: reasonsArray.join(', '),
+        formattedDateSent: format(today, DATE_FORMATS.DD_MMM_YYYY),
+        formattedOldRecords,
+        formattedCorrectRecords: '-',
+        isCompleted: false,
+        bankTeamName: feeRecordCorrectionEntity.bankTeamName,
+        bankTeamEmails: `${bankTeamEmails[0]}, ${bankTeamEmails[1]}`,
+        additionalInfo: feeRecordCorrectionEntity.additionalInfo,
+        formattedBankCommentary: '-',
+        formattedDateReceived: '-',
+      };
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when a correction is completed - isCompleted=true', () => {
+    const feeRecordCorrectionEntity = FeeRecordCorrectionEntityMockBuilder.forIsCompleted(true)
+      .withDateRequested(today)
+      .withBankTeamEmails(`${bankTeamEmails[0]},${bankTeamEmails[1]}`)
+      .build();
+
+    it('should return the fields with formattedDateReceived and formattedBankCommentary as the date and commentary', () => {
+      const result = getRecordCorrectionFields(feeRecordCorrectionEntity.feeRecord, feeRecordCorrectionEntity);
+
+      const { formattedOldRecords, formattedCorrectRecords } = getFormattedOldAndCorrectValues(feeRecordCorrectionEntity, feeRecordCorrectionEntity.feeRecord);
+      const reasonsArray = mapReasonsToDisplayValues(feeRecordCorrectionEntity.reasons);
+
+      const expected = {
+        facilityId: feeRecordCorrectionEntity.feeRecord.facilityId,
+        correctionId: feeRecordCorrectionEntity.id,
+        feeRecordId: feeRecordCorrectionEntity.feeRecord.id,
+        exporter: feeRecordCorrectionEntity.feeRecord.exporter,
+        formattedReasons: reasonsArray.join(', '),
+        formattedDateSent: format(today, DATE_FORMATS.DD_MMM_YYYY),
+        formattedOldRecords,
+        formattedCorrectRecords,
+        isCompleted: true,
+        bankTeamName: feeRecordCorrectionEntity.bankTeamName,
+        bankTeamEmails: `${bankTeamEmails[0]}, ${bankTeamEmails[1]}`,
+        additionalInfo: feeRecordCorrectionEntity.additionalInfo,
+        formattedBankCommentary: feeRecordCorrectionEntity.bankCommentary,
+        formattedDateReceived: format(feeRecordCorrectionEntity.dateReceived as Date, DATE_FORMATS.DD_MMM_YYYY),
```

Comment:
Why do we need `as` here?

Reply:
linting error.  In the controller I have an if exists statement but in the test it causing a linting error

Reply:
Does this work instead?
```suggestion
        formattedDateReceived: format(feeRecordCorrectionEntity.dateReceived!, DATE_FORMATS.DD_MMM_YYYY),
```
If so it is nicer because it is specifically claiming it is not null and so isn't claiming quite as much as `as Date` does

Reply:
Although I actually think it might be even nicer to do:
```
// In the test set up:
const dateReceived = new Date();

 const feeRecordCorrectionEntity = FeeRecordCorrectionEntityMockBuilder.forIsCompleted(true)
      .withDateRecevied(dateReceived)
			.withDateRequested(today)
      .withBankTeamEmails(`${bankTeamEmails[0]},${bankTeamEmails[1]}`)
      .build();

// In the assertion:
formattedDateReceived: format(dateReceived, DATE_FORMATS.DD_MMM_YYYY),
```

Reply:
needs to have ! as otherwise it still causes a lint error.  Even with the additional .withDateRecevied(dateReceived)